### PR TITLE
Display the supported status of a node in skuba cluster status

### PR DESCRIPTION
**Pull request by evrardjp**
This is a first pass at showing the supported state of a cluster,
without adding annotations on the final nodes, by simply relying
on skuba's code. It shows to the users that the cluster is
in an unsupported state, which should help trigger the decision
to upgrade.

## Why is this PR needed?

Does it fix an issue? addresses a business case?

add a description and link to the issue if one exists.

Fixes #

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

please include a brief "management" technical overview (details are in the code)

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: pkg/skuba/actions/cluster/status/status.go [hunk 1]
```diff
@@ -55,19 +57,29 @@ func Status(client clientset.Interface) error {
...
 		"OS-IMAGE:.status.nodeInfo.osImage," +
-		"KERNEL-VERSION:.status.nodeInfo.kernelVersion," +
-		"KUBELET-VERSION:.status.nodeInfo.kubeletVersion," +
+		"KERNEL:.status.nodeInfo.kernelVersion," +
+		"KUBELET:.status.nodeInfo.kubeletVersion," +
 		"CONTAINER-RUNTIME:.status.nodeInfo.containerRuntimeVersion," +
 		"HAS-UPDATES:.metadata.annotations.caasp\\.suse\\.com/has-updates," +
 		"HAS-DISRUPTIVE-UPDATES:.metadata.annotations.caasp\\.suse\\.com/has-disruptive-updates," +
-		"CAASP-RELEASE-VERSION:.metadata.annotations.caasp\\.suse\\.com/caasp-release-version"
+		"RELEASE:.metadata.annotations.caasp\\.suse\\.com/caasp-release-version"
```
Comment (jenting) [inline pkg/skuba/actions/cluster/status/status.go:82]: Remember to update the related readme & documentation.

**Comment by kkaempf** [comment]
This is missing a test case (resp. test description)

**Comment by kkaempf** [comment]
If the intention is to flag the 'zombie' state between a "zypper migration" (SP1 to SP2) and the "skuba migration" (v4 to v5), then 'unsupported' is not expressing the actual status. If a customer gets stuck in the migration, the cluster/node is still supported (in the sense of "you can call support for help").

**Comment by kkaempf** [comment]
That said, flagging the "zombie" state is the right thing to do, so :+1: on the general direction of this PR.

I just wonder if we can find a better wording instead of "unsupported" ?
"unfinished migration" maybe ? "incomplete" ? "updates missing" ?
We could as well tell customers the command that's getting the node out of the zombie state, like `'skuba update' pending` ?!

**Comment by evrardjp** [comment]
Because the stdout already contains numerous details, I didn't want to write extensive details.

We could have something like "Up to date" with values:
- green tick (cluster is up to date);
- warning sign (outdated);
- zombie (unknown state);
- red stop sign (unsupported, too old).

File: pkg/skuba/actions/cluster/status/status.go (hunk 1 above)
Comment (evrardjp) [reply pkg/skuba/actions/cluster/status/status.go:82]: Right :)

**Comment by evrardjp** [comment]
I don't think it would be smart for me to introduce a new logic in this, just before my holidays, if this starts to break. I hope someone can take it away, and add testing. It shouldn't take very long.

**Comment by evrardjp** [comment]
I considered myself that it was a good idea to implement for our users (especially in those migration times), but I have no stake in this game.

If someone wants a feature like this, let's put it in the backlog.

**Comment by jordimassaguerpla** [comment]
> I considered myself that it was a good idea to implement for our users (especially in those migration times), but I have no stake in this game.
> 
> If someone wants a feature like this, let's put it in the backlog.

We would need an issue in github/SUSE/avant-garde to plan for this. Thanks.

**Comment by evrardjp** [comment]
This was just a PoC. If someone wants it, let's make a card for it indeed.

**Pull request closed without merging**